kotlinx-coroutines-core/common/src/internal/Scopes.kt

```diff
@@ -13,8 +13,9 @@ import kotlin.jvm.*
  */
 internal open class ScopeCoroutine<in T>(
     context: CoroutineContext,
-    uCont: Continuation<T>
-) : AbstractCoroutine<T>(context, true, true), CoroutineStackFrame {
+    uCont: Continuation<T>,
+    initParentJob: Boolean
```

Comment:
How about setting a default value?
```suggestion
    initParentJob: Boolean = true
```

Reply:
The precedence in AbstractCoroutine is not to define a default value, and I do think behaviour flags like this should be explicitly supplied by subclasses.